packages/appeals-service-api/src/db/schema.prisma

```diff
@@ -10,8 +7,119 @@ datasource db {
+  // the LPA this user belongs to
+  lpaCode    String?
+  // whether this LPA user is an admin for that LPA
+  isLpaAdmin Boolean?
+  // the status of this user, e.g. have they logged in and confirmed their email?
+  status     String? // added | confirmed | removed
+
+  // appeals where this user is the appellant
+  AppealsAsAppellant Appeal[] @relation("appellant")
+  // appeals where this user is the agent
+  AppealsAsAgent     Appeal[] @relation("agent")
+
+  SecurityToken SecurityToken?
+}
+
+// SecurityToken is used as a mechanism for users to login
+// They enter their email, a token is generated, emailed to the user, and they enter it to login
+// A user only ever has one token, which gets updated when a new token is required
+model SecurityToken {
+  id String @id @default(dbgenerated("newid()")) @db.UniqueIdentifier
+
+  // link to a user
+  appealUserId String     @unique @db.UniqueIdentifier
+  AppealUser   AppealUser @relation(fields: [appealUserId], references: [id])
+
+  // the generated token the user can login with
+  token            String
+  // the time the token was generated, used for checking token expiry
+  tokenGeneratedAt DateTime
+  // the number of attempts at entering the code since this token was last generated
+  // this MUST be updated in a transaction to ensure atomic operation
+  // see https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#atomic-number-operations
+  attempts         Int      @default(0)
+  // the action the code was generated for, e.g. confirmEmail
+  action           String?
+}
+
+// Appeal is the base appeal model, used to link references to other entities, which includes, for example:
+// - an appeal submission authored in the front office
+// - a case published by the back office
+// - an LPA questionnaire submission
+model Appeal {
+  id String @id @default(dbgenerated("newid()")) @db.UniqueIdentifier
+
+  // NOTE
+  // the appellant and agent relations below have onDelete and onUpdate set to NoAction, as SQL can't have multiple cascade paths
+  // to solve this, rather than just one relation havinge NoAction we set both
+  // IF users are deleted, then any relationships will need to be updated
+  // see https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions/special-rules-for-referential-actions
+
+  // link to an AppealUser, who is the appellant
+  // this is optional because we may get an appeal from back office before the associated (service) user entity
+  appellantUserId String?     @db.UniqueIdentifier
+  Appellant       AppealUser? @relation("appellant", fields: [appellantUserId], references: [id], onDelete: NoAction, onUpdate: NoAction)
+
+  // link to an AppealUser, who is the agent
+  // this is optional because not all appeals have an agent
+  agentUserId String?     @db.UniqueIdentifier
+  Agent       AppealUser? @relation("agent", fields: [agentUserId], references: [id], onDelete: NoAction, onUpdate: NoAction)
```

Comment:
According to the PINS data model, `agent` is an array rather than a single entity so I wonder whether we should create a cross-reference table to allow the linking of multiple agents to an appeal. 

If we were to also add a field to the cross reference table for `role`, that would allow us to remove references to all types of users from the appeal table and manage the relationships between entities independently. That would also allow us to manage other types of relation between those entities e.g. `interestedParty`. Obviously the trade-off there is we'd need a table join to retrieve the data rather than querying a single table.

Reply:
Great point! I think a cross reference table is the right solution. I've implemented that.